src/BuiltInTools/AspNetCoreDeltaApplier/StartupHook.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipes;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Watcher.Tools;
+
+internal sealed class StartupHook
+{
+    private static readonly bool LogDeltaClientMessages = Environment.GetEnvironmentVariable("HOTRELOAD_DELTA_CLIENT_LOG_MESSAGES") == "1";
+
+    public static void Initialize()
+    {
+        Task.Run(async () =>
+        {
+            try
+            {
+                await ReceiveDeltas();
+            }
+            catch (Exception ex)
+            {
+                Log(ex.Message);
+            }
+        });
+    }
+
+    private static List<Action> GetAssembliesReceivingDeltas()
+    {
+        var receipients = new List<Action>();
+
+        foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+        {
+            var customAttributes = assembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+            var deltaReceiverAttributes = customAttributes.Where(a => a.Key == "ReceiveHotReloadDeltaNotification" && !string.IsNullOrEmpty(a.Value));
```

Comment:
Who is responsible for applying the `ReceiveHotReloadDeltaNotification` on the app's assemblies. The compiler?

Reply:
This will eventually get replaced by a runtime API: https://github.com/dotnet/runtime/issues/49361. This is a place holder until those APIs come online.